### PR TITLE
GitHub Actions Release Process: signing, artifactory, sonatype

### PR DESCRIPTION
**Pull request by simonbasle**
- add `qualifyVersion` function and `qualifyVersionGha` task﻿ to build
 - conditionally apply signing and artifactory plugins
 - add support for publishing to Maven Central / sonatype for RELEASEs
 - remove push target from CI workflow
 - introduce publish.yml workflow with isolated jobs with different amounts of
 "privileges" via github environments: snapshots, milestones and releases.

### CODE REVIEW

**Comment by simonbasle** [comment]
merge: this depends on the `snapshots` and `releases` github environments to be created for the project before merging. the intent is for the snapshots env to have only artifactory "publish-to-libs-snapshot" credentials and no validation. The release env on the other hand will require approval, having more secrets.

**Review by violetagg — APPROVED** [review]

**Comment by simonbasle** [comment]
see reactor/reactor#694

File: .github/workflows/publish.yml
Comment (simonbasle) [inline .github/workflows/publish.yml:34]: oops this is only for actions with inputs

Comment (simonbasle) [reply .github/workflows/publish.yml:34]: see https://github.com/reactor/reactor-addons/pull/253

**Comment by simonbasle** [comment]
TODO before merging:
 - [x] create the environments
 - [ ] deactivate bamboo builds

**Review by violetagg — APPROVED** [review]

**Pull request merged**
